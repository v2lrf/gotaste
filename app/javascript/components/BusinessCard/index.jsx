import React from 'react'
import PropTypes from 'prop-types'
import { InfoItem } from '../InfoItem'

export function BusinessCard({
  href,
  name,
  streetName,
  streetNumber,
  postalCode,
  city
}) {
  return (
    <a
      href={href}
      className="inline-flex flex-col rounded shadow no-underline hover:shadow-md mb-4"
    >
      <div>
        <img
          src="https://loremflickr.com/512/192"
          alt=""
          className="rounded-t"
        />
      </div>
      <div className="p-4">
        <InfoItem
          imageSrc="https://loremflickr.com/64/64"
          headline={name}
          tagline={`${streetName} ${streetNumber}, ${postalCode} ${city}`}
        />
      </div>
    </a>
  )
}

BusinessCard.propTypes = {
  href: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  streetName: PropTypes.string.isRequired,
  streetNumber: PropTypes.string.isRequired,
  postalCode: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired
}

export default BusinessCard
