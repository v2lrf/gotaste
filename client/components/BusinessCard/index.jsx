import React from 'react'
import PropTypes from 'prop-types'
import Link from 'next/link'
import { Image } from 'cloudinary-react'

import config from '../../config'

import InfoItem from '../InfoItem'

function BusinessCard({ name, address, logoId, heroImageId, slug }) {
  const { streetName, streetNumber, postalCode, city } = address
  return (
    <Link
      href={{ pathname: '/business', query: { slug } }}
      as={`/business/${slug}`}
    >
      <a className="inline-flex flex-col rounded shadow-lg no-underline mb-4 hover-scale">
        <div>
          <Image
            cloudName={config.cloudinaryCloudName}
            publicId={heroImageId}
            height={200}
            width={600}
            crop="fill"
            className="rounded-t"
            secure="true"
          />
        </div>
        <div className="p-4">
          <InfoItem
            imageSrc={logoId}
            headline={name}
            tagline={`${streetName} ${streetNumber}, ${postalCode} ${city}`}
          />
        </div>
      </a>
    </Link>
  )
}

BusinessCard.propTypes = {
  name: PropTypes.string.isRequired,
  address: PropTypes.shape({
    streetName: PropTypes.string.isRequired,
    streetNumber: PropTypes.string.isRequired,
    postalCode: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired
  }).isRequired,
  logoId: PropTypes.string.isRequired,
  heroImageId: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired
}
export default BusinessCard
