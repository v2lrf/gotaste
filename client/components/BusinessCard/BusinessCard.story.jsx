import React from 'react'
import { storiesOf } from '@storybook/react'

import BusinessCard from '.'

const fakeBusiness = {
  name: 'Vinhanen',
  streetName: 'Baggesensgade',
  streetNumber: '13',
  postalCode: '2200',
  city: 'KÃ¸benhavn N'
}

storiesOf('BusinessCard', module).add('Default', () => (
  <div className="p-16">
    <BusinessCard href="#" {...fakeBusiness} />
  </div>
))
