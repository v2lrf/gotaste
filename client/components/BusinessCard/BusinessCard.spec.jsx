import React from 'react'
import BusinessCard from '.'

describe('BusinessCard', () => {
  let component
  let props
  beforeEach(() => {
    props = {
      slug: 'some-path',
      name: 'Fine Wines',
      address: {
        streetName: 'Fake Street',
        streetNumber: '123',
        postalCode: '6660',
        city: 'New Awesome'
      },
      logoId: 'logoId',
      heroImageId: 'heroImageId'
    }

    component = shallow(<BusinessCard {...props} />)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
