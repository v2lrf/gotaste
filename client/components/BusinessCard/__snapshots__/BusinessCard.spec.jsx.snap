// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessCard Snapshots should match the snapshot 1`] = `
<Link
  as="/business/some-path"
  href={
    Object {
      "pathname": "/business",
      "query": Object {
        "slug": "some-path",
      },
    }
  }
>
  <a
    className="inline-flex flex-col rounded shadow-lg no-underline mb-4 hover-scale"
  >
    <div>
      <Image
        className="rounded-t"
        cloudName="dkrjpli3y"
        crop="fill"
        height={200}
        publicId="heroImageId"
        secure="true"
        width={600}
      />
    </div>
    <div
      className="p-4"
    >
      <InfoItem
        headline="Fine Wines"
        imageAlt=""
        imageSrc="logoId"
        large={false}
        tagline="Fake Street 123, 6660 New Awesome"
      />
    </div>
  </a>
</Link>
`;
